apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -696,7 +690,9 @@ def _add_ip_mapping_details(self, context, port, host, owned_addr,
             context,
             filters={'name': self.apic_manager.ext_net_dict.keys()})
         ext_nets = {n['id']: n for n in ext_nets
-                    if self._is_nat_enabled_on_ext_net(n)}
+                    if self._is_nat_enabled_on_ext_net(n) and
+                    not self._is_edge_nat(
+                        self.apic_manager.ext_net_dict.get(n['name']))}
```

Comment:
Why is this a get() instead of ext_net_dict[n['name']]? Besides you are not handling the case where get() returns None. I can see that get() will never return None here (because of the get_networks call above), but then get() should not be needed either.